commit f9ea48c350feb9054c7176c2adec9c8e4a06f503
Author: Jeroen Demeyer <jdemeyer@cage.ugent.be>
Date:   Mon Feb 5 11:24:14 2018 +0100

    Work around https://gcc.gnu.org/bugzilla/show_bug.cgi?id=10200
    
    This shows up in particular on Solaris systems with GCC-5.4.0

diff --git a/fplll/sieve/sieve_gauss_2sieve.cpp b/fplll/sieve/sieve_gauss_2sieve.cpp
index 83eb817..625c6f0 100644
--- a/fplll/sieve/sieve_gauss_2sieve.cpp
+++ b/fplll/sieve/sieve_gauss_2sieve.cpp
@@ -33,7 +33,7 @@ template <class ZT, class F> Z_NR<ZT> GaussSieve<ZT, F>::update_p_2reduce(ListPo
     for (lp_it = List.begin(); lp_it != List.end(); ++lp_it)
     {
       v = *lp_it;
-      if (p->norm < v->norm)
+      if ((p->norm) < v->norm)
         break;
 
       /* if there is one reduction the vector should re-pass the list */
diff --git a/fplll/sieve/sieve_gauss_3sieve.cpp b/fplll/sieve/sieve_gauss_3sieve.cpp
index 3c3716c..3fdbca6 100644
--- a/fplll/sieve/sieve_gauss_3sieve.cpp
+++ b/fplll/sieve/sieve_gauss_3sieve.cpp
@@ -27,7 +27,7 @@ GaussSieve<ZT, F>::update_p_3reduce_2reduce(ListPoint<ZT> *p,
     for (lp_it = List.begin(); lp_it != List.end(); ++lp_it)
     {
       v = *lp_it;
-      if (p->norm < v->norm)
+      if ((p->norm) < v->norm)
         break;
       if (half_2reduce(p, v))
       {
@@ -166,7 +166,7 @@ template <class ZT, class F> Z_NR<ZT> GaussSieve<ZT, F>::update_p_3reduce(ListPo
         continue;
       }
       ++lp_it2;
-      if (v1->norm < p->norm)
+      if ((v1->norm) < p->norm)
       {
         vnew2 = new_listpoint<ZT>(nc);
         if (check_3reduce(v1, p, v2, vnew2) != 1)
diff --git a/fplll/sieve/sieve_gauss_4sieve.cpp b/fplll/sieve/sieve_gauss_4sieve.cpp
index 9eb586e..f2758a4 100644
--- a/fplll/sieve/sieve_gauss_4sieve.cpp
+++ b/fplll/sieve/sieve_gauss_4sieve.cpp
@@ -15,7 +15,7 @@ void GaussSieve<ZT, F>::update_p_4reduce_aux(ListPoint<ZT> *p,
   for (lp_it = List.begin(); lp_it != List.end(); ++lp_it)
   {
     v = *lp_it;
-    if (p->norm < v->norm)
+    if ((p->norm) < v->norm)
       break;
   }
   lp_it_k = lp_it;
@@ -119,7 +119,7 @@ template <class ZT, class F> Z_NR<ZT> GaussSieve<ZT, F>::update_p_4reduce_3reduc
         continue;
       }
       ++lp_it2;
-      if (v1->norm < p->norm)
+      if ((v1->norm) < p->norm)
       {
         /*cout << "#   --- here 1 " << endl;
         cout << v1->norm << endl;
@@ -274,7 +274,7 @@ template <class ZT, class F> Z_NR<ZT> GaussSieve<ZT, F>::update_p_4reduce(ListPo
         }
         ++lp_it3;
         /* (v1, p, v2, v3) or (v1, v2, p, v3) */
-        if (v1->norm < p->norm)
+        if ((v1->norm) < p->norm)
         {
           /* (v1, p, v2, v3) */
           if (v2->norm > p->norm)
